enterprise/server-enterprise/src/main/java/org/neo4j/server/enterprise/EnterpriseServerConfigLoader.java

```diff
@@ -36,7 +36,10 @@
     public Config loadConfig( File configFile, File legacyConfigFile, Log log, Pair<String,String>... configOverrides )
     {
         Config config = super.loadConfig( configFile, legacyConfigFile, log, configOverrides );
-        config.registerSettingsClasses( asList( HaSettings.class, ClusterSettings.class ) );
+        if ( config.get( EnterpriseServerSettings.mode ).equals( "HA" ) )
```

Comment:
I honestly don't think we should do this here - it means we've duplicated this in both the factory and here. Adding it as a line in `HighlyAvailableEditionModule` means we solve the problem for both places, and sets a good precedent for registering settings classes along with the modules they configure.